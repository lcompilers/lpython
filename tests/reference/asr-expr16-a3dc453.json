{
    "basename": "asr-expr16-a3dc453",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/expr16.py",
    "infile_hash": "c87ce02d4b227d695bad17d79c7359c0d8ffd64b5c5c231bce5528e4",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-expr16-a3dc453.stdout",
    "stdout_hash": "8649269b59b3784ecbb13a4d053d83b1cf2fe7b30f770b859e2aaa84",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}