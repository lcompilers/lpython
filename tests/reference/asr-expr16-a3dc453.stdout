(TranslationUnit (SymbolTable 1 {main_program: (Program (SymbolTable 3 {}) main_program [] []), test_issue_455: (Function (SymbolTable 2 {_lpython_return_variable: (Variable 2 _lpython_return_variable ReturnVar () () Default (Real 8 []) Source Public Required .false.), data: (Variable 2 data In () () Default (List (Integer 4 [])) Source Public Required .false.), i: (Variable 2 i Local () () Default (Integer 4 []) Source Public Required .false.), sum: (Variable 2 sum Local () () Default (Real 8 []) Source Public Required .false.)}) test_issue_455 [(Var 2 data)] [(= (Var 2 sum) (RealConstant   0.00000000000000000e+00 (Real 8 [])) ()) (DoLoop ((Var 2 i) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (ListLen (Var 2 data) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(= (Var 2 sum) (RealBinOp (Var 2 sum) Add (Cast (ListItem (Var 2 data) (Var 2 i) (Integer 4 []) ()) IntegerToReal (Real 8 []) ()) (Real 8 []) ()) ())]) (= (Var 2 _lpython_return_variable) (Var 2 sum) ()) (Return)] (Var 2 _lpython_return_variable) Source Public Implementation ())}) [])
